### DISCUSSION TITLE
What is Language Switch (Settings -> Input -> Key mappings)?

### DISCUSSION THREAD

**Discussion by gothka404** [gujjwal00/avnc#268] [General]
In order to to [use the super key](https://github.com/gujjwal00/avnc/issues/264) while keeping Alt key, I tried enabling "Language Switch -> Super".

Currently this doesn't work in a GNOME/Debian environment, with default shortcut Super+Space for "Switch to next input source". It was tried with both a single as well as multiple configured languages.

Am I misinterpreting this setting or how is it supposed to work?

**Comment by gujjwal00** [comment]
[Here](https://developer.android.com/reference/android/view/KeyEvent#KEYCODE_LANGUAGE_SWITCH) is the documentation for Language Switch key. I don't use is personally, it was added by a contributor. 

And this mapping happens _before_ event is sent to server. If AVNC receives a Language Switch key event from Android, it will pretend as if Super was pressed, and send the Super key event to server.

**Comment by gothka404** [comment]
Thanks @gujjwal00. Switching language on Android keyboard didn't work either for me. Hence for now I'll have none of the key mappings activated and don't use the super key as plain key at all. Since I am mainly interested in Activities Overview, there is a an additional shortcut for it, which by default is Super+s.
